File: sdks/insuretech-go-sdk/pkg/models/quote.go
package models

import (
	"time"
)

// Quote represents a quote
type Quote struct {
	SumAssured *Money `json:"sum_assured,omitempty"`
	TermYears int `json:"term_years"`
	PremiumPaymentMode string `json:"premium_payment_mode"`
	BasePremium *Money `json:"base_premium,omitempty"`
	TaxAmount *Money `json:"tax_amount,omitempty"`
	ConvertedPolicyId string `json:"converted_policy_id,omitempty"`
	BeneficiaryId string `json:"beneficiary_id"`
	RiderPremium *Money `json:"rider_premium,omitempty"`
	TotalPremium *Money `json:"total_premium,omitempty"`
	PremiumCalculation string `json:"premium_calculation,omitempty"`
	ApplicantAge int `json:"applicant_age"`
	Smoker bool `json:"smoker,omitempty"`
	ValidUntil time.Time `json:"valid_until"`
	AuditInfo interface{} `json:"audit_info"`
	Status interface{} `json:"status"`
	ConvertedAt time.Time `json:"converted_at,omitempty"`
	Id string `json:"id"`
	QuoteNumber string `json:"quote_number"`
	InsurerProductId string `json:"insurer_product_id"`
	SelectedRiders string `json:"selected_riders,omitempty"`
	ApplicantOccupation string `json:"applicant_occupation,omitempty"`
}

File: backend/insurance_engine-old/src/InsuranceEngine.Underwriting/Domain/Entities/Quote.cs
using System;
using InsuranceEngine.Underwriting.Domain.Enums;
using InsuranceEngine.SharedKernel.Domain.ValueObjects;

namespace InsuranceEngine.Underwriting.Domain.Entities;

public class Quote
{
    public Guid Id { get; set; }
    public string QuoteNumber { get; set; } = string.Empty;
    public Guid BeneficiaryId { get; set; }
    public Guid InsurerProductId { get; set; }
    public QuoteStatus Status { get; set; }

    public long SumAssuredAmount { get; set; }
    public string SumAssuredCurrency { get; set; } = "BDT";
    
    public int TermYears { get; set; }
    public string PremiumPaymentMode { get; set; } = "YEARLY";

    public long BasePremiumAmount { get; set; }
    public string BasePremiumCurrency { get; set; } = "BDT";
    public long RiderPremiumAmount { get; set; }
    public string RiderPremiumCurrency { get; set; } = "BDT";
    public long TaxAmount { get; set; }
    public string TaxCurrency { get; set; } = "BDT";
    public long TotalPremiumAmount { get; set; }
    public string TotalPremiumCurrency { get; set; } = "BDT";

    public string? PremiumCalculationJson { get; set; }
    public string? SelectedRidersJson { get; set; }

    public int ApplicantAge { get; set; }
    public string? ApplicantOccupation { get; set; }
    public bool IsSmoker { get; set; }

    public DateTime ValidUntil { get; set; }
    public Guid? ConvertedPolicyId { get; set; }
    public DateTime? ConvertedAt { get; set; }

    public string? AuditInfoJson { get; set; }

    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
    public DateTime? DeletedAt { get; set; }

    public Money SumAssured => new(SumAssuredAmount, SumAssuredCurrency);
    public Money BasePremium => new(BasePremiumAmount, BasePremiumCurrency);
    public Money RiderPremium => new(RiderPremiumAmount, RiderPremiumCurrency);
    public Money Tax => new(TaxAmount, TaxCurrency);
    public Money TotalPremium => new(TotalPremiumAmount, TotalPremiumCurrency);
}
